src/explore-education-statistics-common/src/modules/table-tool/components/DataTableCaption.tsx

```diff
@@ -58,9 +59,24 @@ const DataTableCaption = ({
     publicationName,
   });
 
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const captionLimit = 300;
   return (
     <>
-      <strong id={id}>{caption}</strong>
+      <strong id={id}>
+        {caption.length > captionLimit && !expanded
+          ? `${caption.substring(0, captionLimit)}...`
+          : caption}{' '}
+      </strong>
+      {caption.length > captionLimit && (
+        <ButtonText
```

Comment:
I don't know if having buttons in captions are screen reader friendly?

Reply:
I ran through it with a screenreader and seems ok :)